### PR TITLE
Add header component

### PR DESCRIPTION
**Pull request by turbo-stump**

### CODE REVIEW

File: src/lib/header/navbar/Navbar.svelte [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		href: string;
+	}[] = [];
+</script>
+
+<nav class="nav">
+	<ul class="nav__list">
+		{#each navLinks as link}
+			{#if isConnected}
+				<NavLinkItem {link} />
+			{:else if link.href !== '/account' && link.href !== '/exchanges'}
```
Comment (stuffy-idler) [inline src/lib/header/navbar/Navbar.svelte:19]: I propose to filter `navLinks` at `<script>` section – it's better to separate UI and custom logic.

File: src/lib/header/navbar/Navbar.svelte [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<script lang="ts" context="module">
+  import NavLinkItem from './NavLinkItem.svelte';
+</script>
+
+<script lang="ts">
+
+	export let isConnected: boolean = false;
+	export let navLinks: {
```
Comment (stuffy-idler) [inline src/lib/header/navbar/Navbar.svelte:8]: You reuse this type (link) everywhere – let's define it at one place and reuse where it needed.

File: tests/header.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { expect, test } from '@playwright/test';
+
+const requiredLinks = ["/", "/account", "/exchanges", "/stats", "https://pepe-team.tawk.help/"]
+
+test('Header should contain required links', async ({ page }) => {
+	await page.goto('/');
+	const linkElements = await page.locator('header nav a');
+	const links = linkElements.evaluateAll(list => list.map(link => link.getAttribute('href')));
+	expect(await links).toEqual(expect.arrayContaining(requiredLinks));
```
Comment (stuffy-idler) [inline tests/header.test.ts:9]: Let's move `await` from `expect` to `links` initializing.

File: src/lib/header/Header.svelte [hunk 4]
```diff
@@ -1,19 +1,45 @@
...
-			</li>
-			<li class:active={$page.url.pathname === '/todos'}>
-				<a sveltekit:prefetch href="/todos">Todos</a>
-			</li>
-		</ul>
-	</nav>
+<header class="header">
+	<div class="header__logo">Logo</div>
+	<NavBar {isConnected} {navLinks} />
+	<div class="header__button">
```
Comment (stuffy-idler) [inline src/lib/header/Header.svelte:40]: Should it be visible only in case when user is not connected?

File: tests/header.test.ts (hunk 3 above)
Comment (turbo-stump) [reply tests/header.test.ts]: done

File: src/lib/header/Header.svelte (hunk 4 above)
Comment (turbo-stump) [reply src/lib/header/Header.svelte:37]: Do you mean header component?
Header component should be visible constantly.
isConnected prop used here for make visible Account and Exchanges links in navbar when user is connected

File: src/lib/header/navbar/Navbar.svelte (hunk 1 above)
Comment (turbo-stump) [reply src/lib/header/navbar/Navbar.svelte]: Maybe this logic to place in upper level in header component? Cause navbar just showing links

Comment (turbo-stump) [reply src/lib/header/navbar/Navbar.svelte]: I made it in last commit

File: src/lib/header/Header.svelte (hunk 4 above)
Comment (turbo-stump) [reply src/lib/header/Header.svelte:45]: fixed earlier

File: src/lib/header/Header.svelte [hunk 5]
```diff
@@ -1,19 +1,50 @@
...
+		}
+	];
+</script>
+
 <script lang="ts">
-	import { page } from '$app/stores';
+	//get value from store
+	let isConnected: boolean = false;
+
+	const getVisibleLinks = (): Link[] => {
```
Comment (stupid-boar) [inline src/lib/header/Header.svelte:35]: => navLinks.filter ..... (without return)

File: src/lib/header/navbar/Navbar.svelte [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<script lang="ts" context="module">
+	import type { Link } from 'src/types/types';
```
Comment (stupid-boar) [inline src/lib/header/navbar/Navbar.svelte:2]: types/types?

File: src/routes/account.svelte [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+	// we don't need any JS on this page, though we'll load
+	// it in dev so that we get hot module replacement...
+	export const hydrate = dev;
+
+	export const router = browser;
+</script>
+
+<svelte:head>
+	<title>Account</title>
```
Comment (stupid-boar) [inline src/routes/account.svelte:12]: P2P - Account

File: src/routes/exchanges.svelte [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<script context="module" lang="ts">
+</script>
+
+<svelte:head>
+	<title>Exchanges</title>
```
Comment (stupid-boar) [inline src/routes/exchanges.svelte:5]: P2P - Exchanges

File: src/routes/exchanges.svelte [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+	<!-- <meta name="description" content="Svelte demo app" /> -->
+</svelte:head>
+
+<section>Exchanges</section>
+
+<style>
+</style>
```
Comment (stupid-boar) [inline src/routes/exchanges.svelte:12]: newline

File: src/routes/index.svelte [hunk 10]
```diff
@@ -1,13 +1,12 @@
 <script context="module" lang="ts">
-	export const prerender = true;
 </script>
 
 <svelte:head>
-	<title>Home</title>
-	<meta name="description" content="Svelte demo app" />
+	<title>pepe 2 pepe</title>
```
Comment (stupid-boar) [inline src/routes/index.svelte:5]: P2P - Marketplace

File: src/routes/stats.svelte [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<script context="module" lang="ts">
+</script>
+
+<svelte:head>
+	<title>Stats</title>
```
Comment (stupid-boar) [inline src/routes/stats.svelte:5]: P2P - Stats

File: src/types/types.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import type { SvelteComponent } from "svelte";
+
+export type Link = {
+  content?: string | SvelteComponent;
+  title?: string;
+  href: string;
+  referrerpolicy?: string;
+  target?: string
+}
```
Comment (stupid-boar) [inline src/types/types.ts:9]: newline

File: tests/header.test.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+test('Nav should route to stats page', async ({ page }) => {
+  await page.goto('/stats');
+  expect(await page.title()).toEqual('Stats');
+})
+
+test('Nav should route to FAQ page', async ({ page }) => {
+  await page.goto('/');
+  await page.goto('https://pepe-team.tawk.help/');
+  expect(await page.title()).toEqual('PepeTeam Help Center');
+})
```
Comment (stupid-boar) [inline tests/header.test.ts:36]: newline

**Review by stupid-boar — CHANGES_REQUESTED** [review]

File: src/lib/header/Header.svelte (hunk 5 above)
Comment (turbo-stump) [reply src/lib/header/Header.svelte:35]: done

File: src/lib/header/navbar/Navbar.svelte (hunk 6 above)
Comment (turbo-stump) [reply src/lib/header/navbar/Navbar.svelte]: moved types.ts to src

File: src/routes/account.svelte (hunk 7 above)
Comment (turbo-stump) [reply src/routes/account.svelte]: done

File: src/routes/exchanges.svelte (hunk 8 above)
Comment (turbo-stump) [reply src/routes/exchanges.svelte]: done

File: src/routes/exchanges.svelte (hunk 9 above)
Comment (turbo-stump) [reply src/routes/exchanges.svelte]: done

File: src/types/types.ts (hunk 12 above)
Comment (turbo-stump) [reply src/types/types.ts]: done

File: tests/header.test.ts (hunk 13 above)
Comment (turbo-stump) [reply tests/header.test.ts]: done

File: src/routes/index.svelte (hunk 10 above)
Comment (turbo-stump) [reply src/routes/index.svelte]: done

File: src/routes/stats.svelte (hunk 11 above)
Comment (turbo-stump) [reply src/routes/stats.svelte]: done

**Review by stupid-boar — DISMISSED** [review]

File: src/lib/header/Header.svelte (hunk 4 above)
Comment (stuffy-idler) [reply src/lib/header/Header.svelte:45]: No, the question was about button "Connect" – when the user is already connected, this button is excess.

File: src/types.ts [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import type { SvelteComponent } from "svelte";
+
+export type Link = {
+  content?: string | SvelteComponent;
+  title?: string;
```
Comment (stuffy-idler) [inline src/types.ts:5]: What is the difference between `content` and `title`? Can it be merged to single field?

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 14 above)
Comment (turbo-stump) [reply src/types.ts:5]: link type fixed

**Review by stuffy-idler — APPROVED** [review]

**Pull request merged**